dynamic_programming/optimal_strategy_for_game.py

```diff
@@ -0,0 +1,74 @@
+"""
+Problem statement:
+
+We are given with N coins having value val_1 ... val_n , where n is even. 
+
+We are playing a game with an opponent in alternating turns. 
+In each particular turm one player selects one or last coin.
+We remove the coin from the row and get the value of the coin.
+
+Find the max possinle amoint of money one can win 
+
+"""
+
+
+#Input: value in form of given sequence of coins
+#Output: Maximum value a player can win from the array of coins
+
+#Imp: N must be even  
+
+def optimal_strategy(coin, N): 
+      
+    #table to store sub problems
+
+ 
```

Comment:
Hi, @Prerna11071997, can you please add type hints and doctests to all the functions as mentioned in the contributing guidelines, before the PR is marked ready for review.